rudder-core/src/main/scala/com/normation/rudder/repository/ldap/LDAPEntityMapper.scala

```diff
@@ -251,7 +251,15 @@ class LDAPEntityMapper(
                           case x :: Nil => Full(x)
                           case _ => Failure(s"Too many policy servers for a Node '${node.id.value}'. Entry details: ${inventoryEntry}")
                         }
-      agentsName  <- sequence(inventoryEntry.valuesFor(A_AGENTS_NAME).toSeq) {AgentInfoSerialisation.parseCompatNonJson}
+      keys =  inventoryEntry.valuesFor(A_PKEYS).map(Some(_))
+
+      agentsName  <- {
+        val agents = inventoryEntry.valuesFor(A_AGENTS_NAME).toSeq.map(Some(_))
+        sequence(agents.zipAll(keys,None,None)) {
+          case (Some(agent),key) => AgentInfoSerialisation.parseCompatNonJson(agent,key)
+          case (None,key) => Failure("Should not happen")
```

Comment:
please use a more descriptive comment that could actually help the user (or dev) understand what was wrong